wgpu-core/src/command/compute.rs

```diff
@@ -251,10 +251,16 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                     // Rebind resources
                     if state.binder.pipeline_layout_id != Some(pipeline.layout_id.value) {
                         let pipeline_layout = &pipeline_layout_guard[pipeline.layout_id.value];
-                        state.binder.pipeline_layout_id = Some(pipeline.layout_id.value);
-                        state
+
+                        let old_pipeline_layout_id = state
```

Comment:
I was thinking more like something `state.binder.change_pipeline_layout(&*pipeline_layout_guard, pipeline.layout_id.value)`, which internally do all the things, including `reset_expectations`